src/emit.c

```diff
@@ -1562,6 +1563,51 @@ static int process_wq_end(struct worker_state *w, struct wprof_event *e, size_t
 	return 0;
 }
 
+static const char *scx_dsq_insert_type_str(enum scx_dsq_insert_type type)
+{
+	switch (type) {
+	case SCX_DSQ_INSERT: return "dsq_insert";
+	case SCX_DSQ_INSERT_VTIME: return "dsq_insert_vtime";
+	case SCX_DSQ_DISPATCH: return "dispatch";
+	case SCX_DSQ_DISPATCH_VTIME: return "dispatch_vtime";
+	case SCX_DSQ_MOVE: return "dsq_move";
+	case SCX_DSQ_MOVE_VTIME: return "dsq_move_vtime";
+	case SCX_DSQ_DISPATCH_FROM_DSQ: return "dispatch_from_dsq";
+	case SCX_DSQ_DISPATCH_VTIME_FROM_DSQ: return "dispatch_vtime_from_dsq";
+	default: return "unknown";
+	}
+}
+
+/* EV_SCX_DSQ_END */
+static int process_scx_dsq_end(struct worker_state *w, struct wprof_event *e, size_t size)
+{
+	if (!should_trace_task(&e->task))
+		return 0;
+
+	(void)task_state(w, &e->task);
+
+	const char *insert_type_str = scx_dsq_insert_type_str(e->scx_dsq.scx_dsq_insert_type);
+	const char *name = sfmt("DSQ:%s_0x%llx", insert_type_str, (u64)e->scx_dsq.scx_dsq_id);
```

Comment:
check process_req_event() and how it uses emit_intern_str() to intern/deduplicate all dynamically generated names. Do you think it's worth doing for SCX_DSQ events as well? What's the probability of having lots (like thousands or more) of dsq_ids? If not high, I'd say we should emit_intern_str(). (I bet AI can follow the pattern very well here)